scrapy/exporters.py

```diff
@@ -144,47 +145,54 @@ def _beautify_newline(self, new_item=False):
         if self.indent is not None and (self.indent > 0 or new_item):
             self.xg.characters('\n')
 
-    def _beautify_indent(self, depth=1):
+    def _beautify_indent(self, breadcrumbs):
         if self.indent:
-            self.xg.characters(' ' * self.indent * depth)
+            self.xg.characters(' ' * self.indent * (len(breadcrumbs) + 1))
 
     def start_exporting(self):
         self.xg.startDocument()
         self.xg.startElement(self.root_element, {})
         self._beautify_newline(new_item=True)
 
     def export_item(self, item):
-        self._beautify_indent(depth=1)
+        self._beautify_indent(breadcrumbs=tuple())
         self.xg.startElement(self.item_element, {})
         self._beautify_newline()
         for name, value in self._get_serialized_fields(item, default_value=''):
-            self._export_xml_field(name, value, depth=2)
-        self._beautify_indent(depth=1)
+            self._export_xml_field((name,), value)
+        self._beautify_indent(breadcrumbs=tuple())
         self.xg.endElement(self.item_element)
         self._beautify_newline(new_item=True)
 
     def finish_exporting(self):
         self.xg.endElement(self.root_element)
         self.xg.endDocument()
 
-    def _export_xml_field(self, name, serialized_value, depth):
-        self._beautify_indent(depth=depth)
-        self.xg.startElement(name, {})
+    def element(self, breadcrumbs, value):
+        if isinstance(breadcrumbs[-1], int):
+            return {'name': self.list_element}
+        assert isinstance(breadcrumbs[-1], str)
```

Comment:
No `assert`s in production code :wink:

Reply:
Just in case: this is a safe use of `assert` in production code: the condition should always be `True`, it’s a way of telling readers (including coverage checkers) that conditions outside the current scope ensure that this is always `True`.